log/logtest/recorder.go

```diff
@@ -67,8 +67,17 @@ type ScopeRecords struct {
 	// SchemaURL of the telemetry emitted by the scope.
 	SchemaURL string
 
-	// Records are the log records this instrumentation scope recorded.
-	Records []log.Record
+	// RecordsWithContext are the log records, and their associated context this
+	// instrumentation scope recorded.
+	RecordsWithContext []RecordWithContext
+}
+
+// RecordWithContext holds a log record the instrumentation received, alongside
+// its context.
+type RecordWithContext struct {
+	log.Record
+
+	Context context.Context
```

Comment:
I think for sake of consistency it would be better to a getter and setter.

Reply:
That makes sense. We only need the getter though, since the package can set the value directly.